c_api/tests/c_api_behavior_test.cpp

```diff
@@ -109,7 +110,554 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
+    ti_destroy_runtime(runtime);
+  };
+  inner(TI_ARCH_VULKAN);
+}
+
+void test_behavior_load_aot_module_impl(TiArch arch) {
+  const auto folder_dir = getenv("TAICHI_AOT_FOLDER_PATH");
+  const std::string module_path = folder_dir + std::string("/module.tcm");
+
+  if (!ti::is_arch_available(arch)) {
+    TI_WARN("arch {} is not supported, so the test is skipped", arch);
+    return;
+  }
+  TiRuntime runtime = ti_create_runtime(arch);
+
+  // AOT module from tcm file, normal usage.
+  {
+    TiAotModule module = ti_load_aot_module(runtime, module_path.c_str());
+    CHECK_TAICHI_SUCCESS();
+    TI_ASSERT(module != TI_NULL_HANDLE);
+  }
+
+  // AOT module from filesystem directory, normal usage.
+  {
+    TiAotModule module = ti_load_aot_module(runtime, folder_dir);
+    CHECK_TAICHI_SUCCESS();
+    TI_ASSERT(module != TI_NULL_HANDLE);
+  }
+
+  // Attempt to load aot module from tcm file, while runtime is null.
+  {
+    TiAotModule module =
+        ti_load_aot_module(TI_NULL_HANDLE, module_path.c_str());
+    CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+    TI_ASSERT(module == TI_NULL_HANDLE);
+  }
+
+  // Attempt to load aot module from tcm file, while runtime is null.
+  {
+    TiAotModule module = ti_load_aot_module(TI_NULL_HANDLE, folder_dir);
+    CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+    TI_ASSERT(module == TI_NULL_HANDLE);
+  }
+
+  // Attempt to load aot module without path.
+  {
+    TiAotModule module = ti_load_aot_module(runtime, nullptr);
+    CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+    TI_ASSERT(module == TI_NULL_HANDLE);
+  }
+  // Attempt to load aot module with a invalid path.
+  {
+    TiAotModule module = ti_load_aot_module(runtime, "ssssss///");
+    CHECK_TAICHI_ERROR_IS(TI_ERROR_CORRUPTED_DATA);
+    TI_ASSERT(module == TI_NULL_HANDLE);
+  }
+  ti_destroy_runtime(runtime);
+}
+
+TEST_F(CapiTest, TestBehaviorLoadAOTModuleVulkan) {
+  test_behavior_load_aot_module_impl(TI_ARCH_VULKAN);
+}
+
+void test_behavior_destroy_aot_module_impl(TiArch arch) {
+  const auto folder_dir = getenv("TAICHI_AOT_FOLDER_PATH");
+  const std::string module_path = folder_dir + std::string("/module.tcm");
+  if (!ti::is_arch_available(arch)) {
+    TI_WARN("arch {} is not supported, so the test is skipped", arch);
+    return;
+  }
+
+  TiRuntime runtime = ti_create_runtime(arch);
+  TiAotModule module = ti_load_aot_module(runtime, module_path.c_str());
+  TI_ASSERT(module != TI_NULL_HANDLE);
+  ti_destroy_aot_module(module);
+  CHECK_TAICHI_SUCCESS();
+  ti_destroy_runtime(runtime);
+
+  // Attempt to destroy a null handle.
+  ti_destroy_aot_module(TI_NULL_HANDLE);
+  CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+}
+
+TEST_F(CapiTest, TestBehaviorDestroyAotModuleVulkan) {
+  test_behavior_destroy_aot_module_impl(TI_ARCH_VULKAN);
+}
+
+void test_behavir_get_cgraph_impl(TiArch arch) {
+  const auto folder_dir = getenv("TAICHI_AOT_FOLDER_PATH");
+  const std::string module_path = folder_dir + std::string("/module.tcm");
```

Comment:
You can simply load it from `folder_dir` then you no longer need to change `graph_aot_test.py`.